aibolit/ast_framework/ast.py

```diff
@@ -67,7 +71,7 @@ def __str__(self) -> str:
                 depth -= print_step
         return printed_graph
 
-    def subtrees_with_root_type(self, root_type: ASTNodeType) -> Iterator[List[int]]:
+    def get_subtrees(self, root_type: ASTNodeType) -> Iterator['AST']:
```

Comment:
```suggestion
    def get_subtrees(self, root_type: ASTNodeType) -> Iterator[AST]:
```

can we use type itself?

Reply:
No, we are in the middle of `AST` class. It is not defined yet. If we would use simply `Iterator[AST]`, then we would get an error, something like "undefined token". To prevent it, we should wrap types in quotes. We should do it only in a case class `A` use itself in its methods as argument or return type.